x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/index.tsx

```diff
@@ -295,12 +301,14 @@ export const CreatePackagePolicyPage: React.FunctionComponent = () => {
         return;
       }
 
+      const packagePolicyPath = getPath('policy_details', { policyId: packagePolicy.policy_id });
+
       if (routeState?.onSaveNavigateTo && policy) {
         const [appId, options] = routeState.onSaveNavigateTo;
 
         if (options?.path) {
           const pathWithQueryString = appendOnSaveQueryParamsToPath({
-            path: options.path,
+            path: wasNewAgentPolicyCreated ? packagePolicyPath : options.path,
```

Comment:
maybe worth adding a comment on the expected behavior? (as adding a first test here seems a lot of complexity)